src/coap_mbedtls.c

```diff
@@ -217,47 +217,43 @@ coap_dgram_write(void *ctx, const unsigned char *send_buffer,
  * Server side PSK callback
  */
 static int psk_server_callback(void *p_info, mbedtls_ssl_context *ssl,
-                  const unsigned char *name, size_t name_len )
+                  const unsigned char *identity, size_t identity_len )
 {
   coap_session_t *c_session = (coap_session_t *)p_info;
-  uint8_t buf[COAP_DTLS_MAX_PSK];
-  size_t psk_len;
   coap_dtls_spsk_t *setup_data;
   coap_mbedtls_env_t *m_env;
+  coap_bin_const_t lidentity;
+  const coap_bin_const_t *psk_key;
+
+  if (c_session == NULL)
+    return -1;
+
+  /* Track the Identity being used */
+  lidentity.s = identity ? (const uint8_t*)identity : (const uint8_t*)"";
+  lidentity.length = identity ? identity_len : 0;
```

Comment:
Here as well (see above).

Reply:
Fixed

Reply:
Actually, there is no guarentee that `identity ` (otherwise, why have `identity_len`) is passed in NULL terminated.  So, code needs to stand.

Reply:
Yes, sorry, I did not read correctly.